### DISCUSSION TITLE
ARM support for cloud-hypervisor.

### DISCUSSION THREAD

**Discussion by rishirsinha** [cloud-hypervisor/cloud-hypervisor#3199] [Q&A] [answered]
I have been trying to run cloud-hypervisor on arm. Trying the instructions listed here:

https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/arm64.md

One thing that was listed right at the top was the requirement for GICv3-ITS device. This is normally not enabled on most arm machines I have had the chance of playing with. Can someone who has already gotten this running on an arm post the specific machine and host OS that they have gotten this working on?

**Comment by likebreath** [comment]
@michael2012z @MrXinWang As the ARM experts/maintainers in our community, would you please help with this issue? Thank you.

**Comment by MrXinWang** [comment, marked as answer]
Hi @rishirsinha, currently the only virtio transport supported by Cloud Hypervisor for AArch64 is the `virtio-pci`, which requires a GICv3-ITS device. That's the reason why we added the first paragraph in the AArch64 document. This is because of the target of the project, i.e.:

```
The project focuses on exclusively running modern, cloud workloads, on top of a limited set of hardware architectures and platforms. 
Cloud workloads refers to those that are usually run by customers inside a cloud provider. 
For our purposes this means modern operating systems with most I/O handled by paravirtualised devices (i.e. virtio), no requirement for legacy devices, and 64-bit CPUs.
```

For legacy interrupt controller such as GICv2 or GICv3 without ITS, I think you could try https://github.com/firecracker-microvm/firecracker.

@michael2012z Do you have any more infomration to add?

**Reply by maburlik** [reply]
Thank you @MrXinWang. Do you know of publicly available hardware that supports GICv3-ITS?

**Reply by MrXinWang** [reply]
Hi @maburlik, not really sure but I think most of the available arm64 server will be equipped with the GICv3-ITS.

**Reply by michael2012z** [reply]
Hi, @maburlik, another option is some kind of development/prototyping board like N1SDP (https://developer.arm.com/tools-and-software/development-boards/neoverse-n1-sdp). It is equipped with GIC-600 which has ITS (https://developer.arm.com/documentation/100336/0106/introduction/components).
